nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/clients/OAuth2Client.java

```diff
@@ -88,28 +126,75 @@ public void setEnabled(boolean enable) {
         this.enabled = enable;
     }
 
-    Map<String, Object> toMap() {
+    public Map<String, Object> toMap() {
         Map<String, Object> doc = new HashMap<>();
         doc.put("clientId", id);
         doc.put("clientSecret", secret);
+        doc.put("redirectURI", redirectURI);
         doc.put("name", name);
         doc.put("enabled", enabled);
         return doc;
     }
 
-    static OAuth2Client fromDocumentModel(DocumentModel doc) {
+    public static OAuth2Client fromDocumentModel(DocumentModel doc) {
         String name = (String) doc.getPropertyValue(OAUTH2CLIENT_SCHEMA + ":name");
         String id = (String) doc.getPropertyValue(OAUTH2CLIENT_SCHEMA + ":clientId");
         String secret = (String) doc.getPropertyValue(OAUTH2CLIENT_SCHEMA + ":clientSecret");
+        String redirectURI = (String) doc.getPropertyValue(OAUTH2CLIENT_SCHEMA + ":redirectURI");
         boolean enabled = (Boolean) doc.getPropertyValue(OAUTH2CLIENT_SCHEMA + ":enabled");
 
-        OAuth2Client client = new OAuth2Client(name, id, secret);
+        OAuth2Client client = new OAuth2Client(name, id, secret, redirectURI);
         client.enabled = enabled;
         return client;
     }
 
-    boolean isValidWith(String clientId, String clientSecret) {
+    /**
+     * @since 9.2
+     */
+    public static boolean isRedirectURIValid(String redirectURI) {
+        if (!redirectURI.startsWith("http")) {
+            return true;
+        }
+        if (redirectURI.startsWith("https") || redirectURI.startsWith("http://localhost")) {
```

Comment:
Maybe be a bit stricter on the `localhost` check, to avoid allowing something like `http://localhost.somecompany.com/`? Not sure if it's worth it.